src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx

```diff
@@ -39,22 +38,23 @@ const PasswordRequirements: UISchemaElementComponent<{
 }> = ({ uischema }) => {
   const { data } = useWidgetContext();
   const {
-    fieldKey,
     header,
     id,
     requirements,
     settings,
     userInfo,
     validationDelayMs,
   } = uischema.options as PasswordRequirementsElement['options'];
-  const password = get(data, fieldKey);
+  const password = 'credentials.passcode' in data
+    ? get(data, 'credentials.passcode')
+    : get(data, 'credentials.newPassword');
 
   const [passwordValidations, setPasswordValidations] = useState<PasswordValidation>({});
 
   const getPasswordStatus = (
     ruleKey: string,
     passwordValidation: PasswordValidation,
-  ): Undefinable<PasswordRequirementStatus> => {
+  ): PasswordRequirementStatus | undefined => {
     if (!passwordValidation) {
```

Comment:
when will this scenario happen?

Reply:
It should be `if (!Object.keys(passwordValidation).length) { ...`(will update) but this will happen when there are no applicable password settings.

Reply:
ohh, it's an internal function, nvm